lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -64,7 +66,14 @@ class ThreadRepositoryImpl extends ThreadRepository {
         sort: sort,
         filter: emailFilter?.filter,
         properties: propertiesCreated);
-
+      if (emailFilter?.filterOption != FilterMessageOption.all && networkEmailResponse.emailList!.isNotEmpty) {
+        _getFirstPage(
```

Comment:
This function returns a `Future`. Why don't you use the result of it. Does it need to use `await` to wait until it's done?

Reply:
In this case, we update the missing cache when filter. So no need `await ` it.